File: DSA/Shreya_Goyal_590015491/Day_17_Question_1.c
#include <stdio.h>

int main() {
    int arr[8], i, target;
    int left, right, mid;
    int found = -1;

    printf("Enter elements in sorted order:\n");
    for(i = 0; i < 8; i++) {
        scanf("%d", &arr[i]);
    }

    printf("Enter the number to search: ");
    scanf("%d", &target);

    left = 0;
    right = 7;

    while(left <= right) {
        mid = (left + right) / 2;

        if(arr[mid] == target) {
            found = mid;
            break;
        }
        else if(target < arr[mid]) {
            right = mid - 1;
        }
        else {
            left = mid + 1;
        }
    }

    if(found != -1) {
        printf("Element found at index %d", found);
    } else {
        printf("-1\n");
    }

    return 0;
}

File: DSA/SakshamKashyap_590014183/day17/day17_Question_1.py
# Implement binary search for finding elements in sorted arrays
my_array = list(map(int, input("Enter the elements (give space between them): ").strip().split()))
target_value = int(input("Enter the number you want to find in array: "))

my_array.sort()  # Sorting the provided array
print(f"Sorted array: {my_array}")

def binary_search(arr, target):
    left = 0
    right = len(arr) - 1
    
    while left <= right:
        mid = (left + right) // 2
        
        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            left = mid + 1  # Search in right half
        else:
            right = mid - 1  # Search in left half
    
    return -1  # Element not found

# Perform binary search
result = binary_search(my_array, target_value)

if result != -1:
    print(f"The index of target value is: {result}")
else:
    print("-1")  # Element is not found in array